### PR TITLE
Feature/update storybook 5 3

### LINKED ISSUE #1774: [Child of #1773] Update Storybook to version 6 [Due Date: 03.08.20]
We dont want to redesign storybook on a old version to avoid doubled work. The updated causes some breaking changes and we want to fix them first.

### PR DESCRIPTION
**Pull request by raphaellueckl**
Fixes #1774

My changes are according to the most recent documentation. We use the very newest versions here. Feel free to adapt, but MAKE SURE that the following stuff works on IE11, otherwise we cannot run it on IE11.

It was a real struggle to get to this point. If you see minor issues, I HIGHLY advise you to merge it first and change the smaller stuff afterwards. If you think there are major issues, please take over the PR and add your ideas.

This project was created with the Storybook-Cli. :)

- `npm run test`
- `npm run build-storybook && cd storybook-static && npx http-server` -> Open IE11 on `http://localhost:8080/`
- `npm start` -> Open IE11 on `http://localhost:6006/`
- Verify that the addons work at all times (Keyboard Shortcut: `D`)
- Make sure that assets like the axa logo and the favicons are displayed in all browsers.

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import '@axa-ch/patterns-library-polyfill';
+import './polyfills-ie11.js';
```
Comment (LucaMele) [inline .storybook/preview.js]: @raphaellueckl why this?

Comment (LucaMele) [reply .storybook/preview.js]: i mean, why not keep our ones?

**Review by danyball — DISMISSED** [review]

File: .storybook/preview.js (hunk 1 above)
Comment (raphaellueckl) [reply .storybook/preview.js]: We keep our ones, but "array.split" or something was not supported. But feel free to reduce them. That part caused a lot of work, but in the end, I never fully optimized it, now that you mention it. Good point.

BUT "our ones" are still there, just within that file. :)

Comment (danyball) [reply .storybook/preview.js]: It is still everything working with just our polyfills. Changed that and clicked trough all our components.

**Comment by danyball** [comment]
JUST with npm start!!

We have 1 error on the console. But everything works. And its just on the first component you are clicking on. 

It seems that this is a open storybook-bug:
https://github.com/storybookjs/storybook/issues/10204

The error is gone if we remove `changelog` and `codepreview` addons.

**Comment by danyball** [comment]
JUST with npm start!!

We have a second warning - but it is not reproducable:
`Warning: unmountComponentAtNode(): The node you're attempting to unmount was rendered by React and is not a top-level container. Instead, have the parent component update its state and rerender in order to remove this component.`

File: .storybook/preview.js (hunk 1 above)
Comment (danyball) [reply .storybook/preview.js]: I have splitted that polyfill files into 2 imports:
- the additional polyfills for storybook 5.3
- our good old PL polyfills for our components

File: .storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    addons: [
+      'storybook-readme/register',
+      './.storybook/addons/changelog/register.js',
+      '@storybook/addon-knobs/register',
+      './.storybook/addons/codepreview/register.js',
+      '@storybook/addon-a11y/register'
+    ],
+};
```
Comment (01231) [inline .storybook/main.js:9]: New line?

**Comment by danyball** [comment]
I suggest to update directly to the new SB 6, which will be releases at the end of july (https://github.com/storybookjs/storybook/issues/9311). Because there is a problem at a SB internal file with the 5.3. 

Have installed the 6-RC and it works with just our PL polyfills. No extra pf needed. We just have to some extra work on our SB plugins. There are breaking changes on how to load them. But the showcases are working:
image


We can go these 2 ways:

1. Merge this PR with SB5.3 and the extra polyfills. After upgrading to 6 remove the extra polyfills.
2. Keep the branch of this PR as a basis for SB6 upgrade.

What do you think @NoahLiechti @LucaMele ?

**Comment by 01231** [comment]
I would choose 1. because we have some tasks for the new storybook/welcomepage design and I think the  design tool integration which storybook provides with version 5.3 would be helpful for these tasks.

**Comment by LucaMele** [comment]
@danyball I would wait to the one that works without polyfill hacks. End of July is only in 2 weeks and everyone is on holiday anyway @NoahLiechti so i guess it wont hurt the designer if they have to wait 2 weeks in holiday times for a cleaner setup

**Comment by danyball** [comment]
We decided to not merge this PR (see comments). All checks have passed on this PR right now. So, the branch is ready to upgrade to 6.X.

We are closing the PR but do not delete the branch.

**Pull request closed without merging**